Fixed IllegalArgumentException + refactoring

dmatej opened the pull request:
* The fix is just this: https://github.com/eclipse-ee4j/glassfish/commit/3b4e132a81fd8b3a3de44b92f44146c6e38decd2
* When I tried to find the cause (log message with unresolved parameters) I touched some other things
  * renamed habitat in the naming manager
  * fixed few warnings in DOL
  * LogDomains printed warning about missing (and unused) resource bundle for the SecurityContainer
  * Originally I suspected JarArchive to make some silly exception when trying to resolve the file name. I was wrong, but I added two unit tests and did some trivial cleanup.
* These two commits were originally created in different branch, but I think they should go to master first.

dmatej committed:
IllegalArgumentException in toString - bundle without descriptor is valid

- caused IllegalArgumentException in Application.toString
- toString must not throw exceptions, it is unusable for logging otherwise

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
NPE in toString - refactoring, code cleanups

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

# Conflicts:
#	appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/impl/GlassfishNamingManagerImpl.java

pzygielo commented on appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/Application.java:
```diff
@@ -1249,7 +1250,7 @@ public boolean isUniqueIdSet() {
      */
     public void setUniqueId(long id) {
-        LOG.log(Level.FINE, "[Application] " + getName() + " , uid: " + id);
+        LOG.log(Level.FINE, () -> "setUniquiId: [Application] " + getName() + " , uid: " + id);
```
`setUniquiId` -> `setUniqueId`

dmatej replied:
Oh no, I did it again!

arjantijms commented on appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/impl/GlassfishNamingManagerImpl.java:
```diff
@@ -684,7 +684,7 @@ private ArrayList<SimpleJndiName> listNames(SimpleJndiName name) throws NamingEx
     private String getComponentId() throws NamingException {
         final ComponentInvocation ci;
```
At some point (maybe not now) make these things "componentInvocation" instead of "ci", etc,

dmatej replied:
I agree, but then I would have to go through the whole module as this is "a convention" here if I remember right.
But I will take a look once again in the morning, I already made a typo.

dmatej committed:
Removed catching Throwable from getComponentId

- It was not thrown, invocation.getComponentId() is just a getter
- NamingException doesn't need to be wrapped by another.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Better log message in setUniqueId

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
